drivers/scheduler/k8s/k8s.go

```diff
@@ -7517,3 +7528,29 @@ func ClusterVersion() (string, error) {
 	}
 	return strings.TrimLeft(ver.String(), "v"), nil
 }
+
+func createClonedStorageClassIfRequired(originalStorageClass *storageapi.StorageClass) (string, error) {
+	cloneStorageClass := originalStorageClass
+	clonedSCName := cloneStorageClass.Name + "-clone"
+	clonedSCobj, err := k8sStorage.GetStorageClass(clonedSCName)
+	if err == nil {
+		return clonedSCobj.Name, nil
+	} else if strings.Contains(err.Error(), fmt.Sprintf("storageclasses.storage.k8s.io \"%v\" not found", clonedSCName)) {
+		log.Infof("Cloned SC with the name %v does not exist and will be created", clonedSCName)
+
+		cloneStorageClass.Name = clonedSCName
+		immediate := storageapi.VolumeBindingImmediate
+		cloneStorageClass.VolumeBindingMode = &immediate
+		cloneStorageClass.ResourceVersion = ""
+		cloneStorageClass.UID = ""
+
+		clonedSCobj, err := k8sStorage.CreateStorageClass(cloneStorageClass)
+		if err != nil {
+			log.Infof("Error occurred while creating cloned storage class:- %v", err)
+			return "", err
+		}
+		return clonedSCobj.Name, nil
+	} else {
```

Comment:
This `else` case should resolve errors other than "not found", which I suppose is likely network issues. 
We should at least automatically retry for a few times with some wait mechanism before returning error.

Reply:
I have added a retry mechanism.